script/tool/test/gradle_check_command_test.dart

```diff
@@ -216,4 +254,119 @@ dependencies {
       ]),
     );
   });
+
+  test('fails when namespace does not match AndroidManifest.xml', () async {
```

Comment:
see comment above, also I think part of the value android is trying to put forward is to separate the package from the namespace so that tests can have a custom namespace.